crates/swc_ecma_minifier/src/compress/optimize/inline.rs

```diff
@@ -408,8 +409,21 @@ where
     }
 
     /// Check if the body of a function is simple enough to inline.
-    fn is_fn_body_simple_enough_to_inline(&self, body: &BlockStmt, param_count: usize) -> bool {
-        let cost_limit = 3 + param_count * 2;
+    fn is_fn_body_simple_enough_to_inline(
+        &self,
+        body: &BlockStmt,
+        param_count: usize,
+        usage: &VarUsageInfo,
+    ) -> bool {
+        let param_cost = param_count * 2;
+        // if it's passed as value but not called, the function expr cannot be removed
```

Comment:
If this is the case, couldn't we just return false and skip the work?

Reply:
```js
function foo() {
  g()
}

foo()

module.exports = foo
```
In this case `foo` cannot be removed, but still can be inlined